src/main/java/io/jenkins/plugins/bitbucketpushandpullrequest/BitBucketPPRTrigger.java

```diff
@@ -160,19 +172,45 @@ private void scheduleJob(BitBucketPPRTriggerCause cause, BitBucketPPRAction bitb
     };
     LOGGER.info("Check if job should be triggered due to changes in SCM");
 
-    if (pJob.scheduleBuild2(5, new CauseAction(cause), bitbucketAction) != null) {
-      try {
-        String name = " #" + job.getNextBuildNumber();
-        LOGGER.info(() -> "SCM changes detected in " + job.getName() + ". Triggering " + name);
-      } catch (NullPointerException e) {
-        LOGGER.info(e.getMessage());
-      }
+    QueueTaskFuture<?> future = pJob.scheduleBuild2(5, new CauseAction(cause), bitbucketAction);
+    LOGGER.info(() -> "SCM changes detected in " + job.getName() + ". Triggering " + " #"
+        + job.getNextBuildNumber());
+
+    if (future != null) {
+      handleFuture(bitbucketAction, scmTrigger, future);
     } else {
       LOGGER
           .info(() -> "SCM changes detected in " + job.getName() + ". Job is already in the queue");
     }
   }
 
+  private void handleFuture(BitBucketPPRAction bitbucketAction, SCM scmTrigger,
+      QueueTaskFuture<?> future) {
+    try {
+      Run<?, ?> run = (Run<?, ?>) future.get();
+      Result result = run.getResult();
+      LOGGER.info(() -> "Result is " + result);
+
+      GitSCM gitSCM = (GitSCM) scmTrigger;
+      UserRemoteConfig config = gitSCM.getUserRemoteConfigs().get(0);
+      String url = run.getResult() == Result.SUCCESS ? bitbucketAction.getLinkApprove()
+          : bitbucketAction.getLinkDecline();
+      
+      if (url != null) {
+        String payload = run.getResult() == Result.SUCCESS ? "{ \"approved\": true }"
+            : "{ \"approved\": false }";
```

Comment:
Wouldn't this revoke an previously existing approval on failure? I think it's reasonable but someone might object / want it to be configurable. Anyway it will do for a first shot!